src/PowerShellEditorServices/Services/PowerShell/Host/EditorServicesConsolePSHostUserInterface.cs

```diff
@@ -22,6 +22,8 @@ internal class EditorServicesConsolePSHostUserInterface : PSHostUserInterface
 
         private readonly PSHostUserInterface _consoleHostUI;
 
+        private readonly HashSet<(long, ProgressRecord)> _currentProgressRecords = new();
```

Comment:
`ProgressRecord.GetHashCode` is just the default object impl, so in most cases we'd be storing a `ProgressRecord` for every update sent. If the goal is to avoid the previous work around regarding the dummy string, a `Dictionary<long, ProgressRecord>` is probably needed.

Reply:
I'd actually just stored the tuple cause we need both items later, the `long` as the source ID and the record to mark as completed. Should be fine as-is in a hash set right? I'm not sure if the source IDs are unique so they may not work well as keys in a dictionary.

Reply:
Fixed up!